mindsdb/integrations/handlers/salesforce_handler/salesforce_handler.py

```diff
@@ -0,0 +1,199 @@
+        self.is_connected = False
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self) -> StatusResponse:
+        """
+        Set up the connection required by the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+        if self.is_connected is True:
+            return StatusResponse(True)
+
+        self.connection = sf.client(
+            'salesforce',
+            sf_username=self.connection_data['user'],
+            sf_password=self.connection_data['password'],
+            sf_security_token=self.connection_data['security_token']
+        )
+        self.is_connected = True
+
+        return self.connection
+
+    def disconnect(self):
+        self.connection = None
+        self.is_connected = False
+
+        return self.connection
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Check the Salesforce connection
+        """
+        response = StatusResponse(False)
+        need_to_close = self.is_connected is False
+        if need_to_close is True:
+            self.connect()
+        try:
+            self.connection.query("SELECT Id FROM Account LIMIT 1")
+            response = StatusResponse(True)
+        except Exception as e:
+            response = StatusResponse(False, error=str(e))
+        if need_to_close is True:
+            self.disconnect()
+        return response
+
+    def native_query(self, query: str) -> Response:
+        """
+        Receive SOQL query from Salesforce and return the result as a pandas dataframe
+        Args: 
+            query (str): SOQL query
+        Returns:
+            HandlerResponse
+        """
+        need_to_close = self.is_connected is False
+        connection = self.connection
+
+        if need_to_close is True:
+            self.connect()
+            connection = self.connection
+
+        try:
+            result = connection.select_object_content(
+                sf_instance=self.connection_data['sf_instance'],
+                reportId=self.connection_data['reportId'],
+                export='?isdtp=p1&export=1&enc=UTF-8&xf=csv',
+                sfUrl=sf_instance + reportId + export,
+                response=requests.get(sfUrl, headers=headers)
+                download=response.content.decode('utf-8'),
+                df=pd.read_csv(StringIO(download))
+            )
+            response = Response(RESPONSE_TYPE.SUCCESS, result)
+        except Exception as e:
+            response = Response(RESPONSE_TYPE.ERROR, str(e))
+
+        if need_to_close is True:
+            self.disconnect()
+
+        return response
+
+    def query(self, query: ASTNode) -> Response:
+        """
+        Receive a query from MindsDB and return the result as a pandas dataframe
+        Args:
+            query (ASTNode): MindsDB query
+        Returns:
+            HandlerResponse
+        """
+        renderer = SqlalchemyRender(Salesforce)
```

Comment:
Here we need to provide the dialect, is this working?

Reply:
ahhh so it would be ("salesforce")... I'll test it now